chainer/variable.py

```diff
@@ -776,16 +776,16 @@ def to_intel64(self):
         intel64.check_ideep_available()
         data = self.data
         if data is not None:
-            if isinstance(data, numpy.ndarray):
-                # numpy.ndarray to ideep
-                self._data = [
-                    intel64.ideep.array(
-                        data, itype=intel64.ideep.wgt_array)]
-            elif isinstance(data, cuda.ndarray):
-                # cupy.ndarray to ideep
-                self._data = [
-                    intel64.ideep.array(
-                        data.get(), itype=intel64.ideep.wgt_array)]
+            if isinstance(data, cuda.ndarray):
+                # cupy.ndarray to numpy.ndarray
+                data = data.get()
+            if (isinstance(data, numpy.ndarray) and
+                    intel64.inputs_all_ready((data,))):
+                # numpy.ndarray to ideep.mdarray
+                data = intel64.ideep.array(
+                    data, itype=intel64.ideep.wgt_array)
+            self._data = [data]
```

Comment:
Hmm, I think it should be `self.data = data`, because `VariableNode._data` should be updated as well. Maybe another issue, though.

Reply:
I guess the indentation level of this block is wrong...?

* https://github.com/chainer/chainer/blob/v4.0.0rc1/chainer/variable.py#L762-L765
* https://github.com/intel/chainer/blob/8fe5698d6e8e34675fcbf75ce721107d7c27c706/chainer/variable.py#L737-L740

I'll fix this in separate PR.

Reply:
Oh, I see...